src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -219,28 +230,112 @@ ExpressionSyntax initializer
 				immutable = false;
 			}
 
-			var stuff = GetStuffToCheckForMember(
-				type,
-				typeSyntax,
-				initializer
-			);
+			// Check all initializations of the member for immutability
+			foreach( ExpressionSyntax initializer in initializers ) {
+				var stuff = GetStuffToCheckForMember(
+					type,
+					typeSyntax,
+					initializer
+				);
 
-			if ( stuff == null ) {
-				// null is a signal that there is nothing further that needs to
-				// be checked
-				return immutable;
-			}
+				if( stuff == null ) {
+					// Null is a signal that there is nothing further that needs to
+					// be checked
+					continue;
+				}
 
-			var (typeToCheck, checkKind, getLocation) = stuff.Value;
+				var (typeToCheck, checkKind, getLocation) = stuff.Value;
+
+				if( m_context.IsImmutable(
+					typeToCheck,
+					checkKind,
+					getLocation,
+					out var diagnostic
+				) ) {
+					continue;
+				}
 
-			if( !m_context.IsImmutable( typeToCheck, checkKind, getLocation, out var diagnostic ) ) {
 				diagnosticSink( diagnostic );
 				immutable = false;
 			}
 
 			return immutable;
 		}
 
+
+		/// <summary>
+		/// A readonly field or property may be re-initialized in constructors.
+		/// This method finds all initializations of the field/property.
+		/// </summary>
+		/// <param name="symbol">The field/property to search for</param>
+		/// <param name="initialInitializer">The initialization with the declaration</param>
+		/// <returns>An immutable list of initializations</returns>
+		private ImmutableArray<ExpressionSyntax> GetInitializers(
+			ISymbol symbol,
+			ExpressionSyntax initialInitializer
+		) {
+			List<ExpressionSyntax> initializers = new() {initialInitializer};
+
+			// If the containing symbol isn't a named type (class) then there
+			// will be no constructors
+			if( symbol.ContainingSymbol is not INamedTypeSymbol namedTypeSymbol ) {
+				return initializers.ToImmutableArray();
+			}
+
+			// Iterate through all constructors in the class
+			foreach( IMethodSymbol methodSymbol in namedTypeSymbol.Constructors ) {
+				// Make sure the syntax of the constructor exists
+				if( methodSymbol.DeclaringSyntaxReferences.Length == 0 ) {
+					continue;
+				}
+
+				// Retrieve the constructor syntax
+				var methodSyntax = methodSymbol.DeclaringSyntaxReferences
+				                               .Single()

```

Comment:
We don't align `.`s like this iirc

Reply:
If you put `.DeclaringSyntaxReferences on the next line then they will line up, that's pretty common.

Reply:
The IDE does. I've never seen it done a different way.

Reply:
Resharper? Mine doesn't 🤔